frontend/components/MainPage/sections/AlertSections/LowFunds/LowFunds.tsx

```diff
@@ -1,11 +1,75 @@
+import { round } from 'lodash';
+import { useMemo } from 'react';
+
+import { LOW_AGENT_SAFE_BALANCE } from '@/constants/thresholds';
+import { useMasterBalances } from '@/hooks/useBalanceContext';
+import { useNeedsFunds } from '@/hooks/useNeedsFunds';
+import { useServices } from '@/hooks/useServices';
+import { useStakingProgram } from '@/hooks/useStakingProgram';
+import { useStore } from '@/hooks/useStore';
+
+import { EmptyFunds } from './EmptyFunds';
 import { LowOperatingBalanceAlert } from './LowOperatingBalanceAlert';
 import { LowSafeSignerBalanceAlert } from './LowSafeSignerBalanceAlert';
 import { MainNeedsFunds } from './MainNeedsFunds';
 
-export const LowFunds = () => (
-  <>
-    <MainNeedsFunds />
-    <LowOperatingBalanceAlert />
-    <LowSafeSignerBalanceAlert />
-  </>
-);
+export const LowFunds = () => {
+  const { storeState } = useStore();
+
+  const { selectedAgentConfig } = useServices();
+  const { selectedStakingProgramId } = useStakingProgram();
+  const { isLoaded: isBalanceLoaded, masterEoaNativeGasBalance } =
+    useMasterBalances();
+
+  const { nativeBalancesByChain, olasBalancesByChain, isInitialFunded } =
+    useNeedsFunds(selectedStakingProgramId);
+
+  const chainId = selectedAgentConfig.evmHomeChainId;
+
+  // Check if the safe signer balance is low
+  const isSafeSignerBalanceLow = useMemo(() => {
+    if (!isBalanceLoaded) return false;
+    if (!masterEoaNativeGasBalance) return false;
+    if (!storeState?.isInitialFunded) return false;
+
+    return masterEoaNativeGasBalance < LOW_AGENT_SAFE_BALANCE;
+  }, [isBalanceLoaded, masterEoaNativeGasBalance, storeState]);
+
+  // Show the empty funds alert if the agent is not funded
+  const isEmptyFundsVisible = useMemo(() => {
+    if (!isBalanceLoaded) return false;
+    if (!olasBalancesByChain) return false;
+    if (!nativeBalancesByChain) return false;
+
+    // If the agent is not funded, <MainNeedsFunds /> will be displayed
+    if (!isInitialFunded) return false;
+
+    if (
+      round(nativeBalancesByChain[chainId], 2) === 0 &&
+      round(olasBalancesByChain[chainId], 2) === 0 &&
```

Comment:
curious why you're rounding here 🤔

Reply:
no no, it was more about rounding in the condition, if it works as expected that's fine, just being mindful about whether the rounding would cause a bug or something